Add new vector icon

the-eclectic-dyslexic opened the pull request:
fixes #517 

user themes off
![notheme](https://github.com/Helium314/HeliBoard/assets/112126910/af45e4e5-093d-44f8-ae89-9210f9689ef5)

user themes on
![theme](https://github.com/Helium314/HeliBoard/assets/112126910/3c0a540c-366d-4e53-b57d-dbad944ee1b4)

Tested in android studio using an emulator for (Android 14) and fairphone 4 (Android 13)

Helium314 commented:
Thanks!
What are the `asstes/launcher_icon` files used for? As far as I know, Andorid doesn't support svg files directly.
The playstore icon seems unnecessary too, but it doesn't really matter as it should not be added to the apk anyway.

Helium314 commented:
Oh, and could you also update https://github.com/Helium314/HeliBoard/blob/main/fastlane/metadata/android/en-US/images/icon.png?

the-eclectic-dyslexic commented:
@Helium314 The assets/launcher_icon is the source svg in case you want to reimport/modify/whatever. Would you like them somewhere else?

Sure I can update that. I am just talking with the designer a little, making sure he is happy to ship the changes we have been talking about, to improve readability, and once that is wrapped up I will make sure to fix all those problems along with updating the visuals. I can remove the playstore icon if you like, and I will make sure to update the fastlane icon.

the-eclectic-dyslexic commented:
I hope that is an improvement! 

Let me know if I missed anything, or there are any additional concerns.

the-eclectic-dyslexic commented:
And Just in case you haven't been following the issue where this was discussed with the artist, here are some screen shots of the current build, tested the same way the the first commit.
![theme](https://github.com/Helium314/HeliBoard/assets/112126910/8d83f0a9-24e5-43e0-a86a-da4ccee83860)
![no_theme](https://github.com/Helium314/HeliBoard/assets/112126910/862f8e05-caa2-4aa6-9af7-3d2bc7eeedbe)

Helium314 commented:
Thanks for the update!
I think this is definitiely an improvement over the previous style.

BlackyHawky commented:
This is just a detail, but don't forget to update the Credits section of the Readme with your name and the artist's name. 😉
Can't wait to test on my devices!


Perhaps the license headers of new files can be updated by adding :
`<!--
    SPDX-License-Identifier: GPL-3.0-only
-->`

What do you think Helium?

Helium314 commented:
Actually the license should be choice of the artist (but compatible, obviously).
@FabianOvrWrt maybe some Creative Commons license?

FabianOvrWrt commented:
> Actually the license should be choice of the artist (but compatible, obviously). @FabianOvrWrt maybe some Creative Commons license?

Yeah, I was thinking about http://creativecommons.org/licenses/by-sa/4.0/ which basically means "as long as you credit me, you can use or modify the work as you please".

FabianOvrWrt commented:
Also, checking the PR I still see the font is too thin (not 750 in weight as I proposed). Here's an updated svg with the proper proportions I was going for in case @the-eclectic-dyslexic couldn't figure out my mess last time:

![HeliBoard_4](https://github.com/Helium314/HeliBoard/assets/6712155/3c399343-c00e-4598-a6ba-5eec5f60b938)

the-eclectic-dyslexic commented:
Okay, I should have that done in an few hours. Then hopefully heliboard can have a shiny new hood ornament. 🙂

The only things to keep in mind are the export svgs won't automatically have the license referenced in the comments and if anyone saves the source svg in inkscape as a "minimized svg" the license will disappear from anywhere I put it, unless I literally make it an element floating off canvas. 

I will put it in a comment, and meta data of the source svg for now. I will also put a copy of the creative commons license in the icon_launcher folder, and update the readme accordingly.

Sorry about that @FabianOvrWrt I did check the font weight, I think I just misread the UI? I'll try to figure out what happened there.

the-eclectic-dyslexic commented:
I figured out the font issue. I had the static fonts installed, not the dynamic weight ones. I used the font weight annotation included, but as a consequence of having the wrong version of the font installed it did precisely nothing. The silly thing is I had a passing suspicion about that, but just figured "nah!". My bad! :dizzy_face: 

I have used the version where the text is forced to paths this time, to avoid any poor unsuspecting soul falling into the same trap in the future.

I updated the readme to say the icon is by Fabian "with contributions from" myself. I hope you find that wording accurate @FabianOvrWrt 

I also added a link in the readme to an included CC-BY-SA-4.0 plaintext license and why it is there. The creative commons suggests linking directly to https://creativecommons.org/licenses/by-sa/4.0/ so I included that link in the readme as well.

The SVGs should carry forward their licenses, both in the SPDX identifier and the meta data, unless someone explicitly removes them. It should even carry forward automatically when someone exports SVGs for the layers to import into android studio.

the-eclectic-dyslexic commented:
![notheme](https://github.com/Helium314/HeliBoard/assets/112126910/e59c7009-6e23-4f1c-92a4-5aa1e235243b)
![theme](https://github.com/Helium314/HeliBoard/assets/112126910/11b75f89-04ad-4b41-9068-b746d7179c57)

the-eclectic-dyslexic commented:
The only other thing I can suggest here, is if you want I can compress the commits here to avoid adding so many binaries into the project history. If it doesn't bother anyone though, then I won't bother.

the-eclectic-dyslexic commented:
The only other thing I can suggest here is I could squash the commits to prevent any files from not having licenses at any point in the project history, and it would also get rid of some binaries.

If the record is more important though, or no one minds either way, then I won't bother.

Helium314 commented:
Thanks a lot @the-eclectic-dyslexic and @FabianOvrWrt 
I usually squash PR commits anyway, unless there is a good reason not to.
